### PR TITLE
feat: All edges options working for Single Channel Mode in Logic Analyzer

### LINKED ISSUE #1051: Only "Every Edge" mode is working in Logic Analyzer
**Actual Behaviour**

Only "Every Edge" mode is working in One Channel Mode Logic Analyzer

**Expected Behaviour**

All the options of selecting edges should work

**Steps to reproduce it**

Go to PR #1044 -> Download apk -> Go to Logic Analyzer -> Analyze all the edges option

**LogCat for the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #1051 

**Changes**: Added the functionality of plotting data with different edges option as selected by user

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2108095/app-debug.zip)

NOTE: Please merge PR #1044 before merging this

### CODE REVIEW

**Comment by harsh-2711** [comment]
@CloudyPadmal Please add WIP tag as EVERY FOURTH RISING option is yet to be implemented. Till then you can check for other edges.

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove WIP tag. I have implemented the missing functionality. Please review.

**Comment by Vikum94** [comment]
When you update the apk, it's better to edit at the **APK for testing**

**Comment by harsh-2711** [comment]
Noted. But I thought it would be good to notify the changes.

**Comment by CloudyPadmal** [comment]
@harsh-2711 may be we can upload the APK to comment and then update the link by pasting it in main PR message body?

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove conflicts tag.

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 1]
```diff
@@ -2903,7 +2903,7 @@ public double getPV3() {
     public double getPCS() {
         return this.dac.getVoltage("PCS");
     }
-
+
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Shall we revert this change? Please don't modify other files which you are not addressing in a PR.

**Comment by CloudyPadmal** [comment]
@harsh-2711 please resolve the conflict. Then this is good to be merged!

**Comment by harsh-2711** [comment]
@CloudyPadmal Conflicts removed.

**Comment by CloudyPadmal** [comment]
@Vikum94 @abhinavraj23 @Avjeet guys please review this

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove conflicts tag. @Vikum94 @abhinavraj23 @Avjeet Please review.

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 2]
```diff
@@ -390,7 +397,9 @@ public void onNothingSelected() {
 
     }
 
-    private void updateLogicLines(double[] xData, double[] yData) {
+    // Single Channel Mode
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Can we put this comment closer to the method you are referring to?

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 3]
```diff
@@ -434,6 +443,42 @@ private void updateLogicLines(double[] xData, double[] yData) {
...
+        int check = xaxis;
+        int count = 0;
+
+        for (int i = 1; i < xData.length; i++) {
+            xaxis = (int) xData[i];
+            if (xaxis != check) {
+                if (count == 3) {
+                    tempInput.add(new Entry(xaxis, 0));
+                    tempInput.add(new Entry(xaxis, 1));
+                    tempInput.add(new Entry(xaxis, 0));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Are we adding the same entry again?

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 4]
```diff
@@ -584,15 +650,29 @@ protected void onPostExecute(Void aVoid) {
...
                 double[] yaxis = digitalChannel.getYAxis();
 
-                String str = "";
+                String str1 = "";
+                String str2 = "";
                 for (int i = 0; i < xaxis.length; i++) {
-                    str += String.valueOf(xaxis[i]);
-                    str += " ";
+                    str1 += String.valueOf(xaxis[i]);
+                    str2 += String.valueOf(yaxis[i]);
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Can you have a look at Java's `StringBuilder` class? It's specially designed for String concatenation.

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java (hunk 3 above)
Comment (harsh-2711) [reply app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Yes. To avoid triangular graph.

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java (hunk 2 above)
Comment (harsh-2711) [reply app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: This comment is common for all the three methods. There will be more such comments for every channel mode methods

Comment (CloudyPadmal) [reply app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Yes, my point is not comments. there is an empty line. Please remove it.
